### PR TITLE
[FSDE-1082] Text field height default changes and InfoIcon for form field elements

### PR DESCRIPTION
**Pull request by riya-abraham01**

### CODE REVIEW

File: src/components/Checkbox/Checkbox.js [hunk 1]
```diff
@@ -26,6 +26,13 @@ export default class Checkbox extends React.Component {
       }
     };
 
+    const info = this.props.info ?
+    <div className="chi-label__help">
```
Comment (janibolkvadze) [inline src/components/Checkbox/Checkbox.js:30]: Is this indentation correct? The linter might throw errors when trying to push to Dev / Prod

File: src/components/Checkbox/Checkbox.js [hunk 2]
```diff
@@ -26,6 +26,13 @@ export default class Checkbox extends React.Component {
...
+      <button className="chi-button -icon -sm -flat" id="example__help-button" aria-label="Help" data-target="#example__help-popover">
```
Comment (janibolkvadze) [inline src/components/Checkbox/Checkbox.js:31]: What if this element is used twice on the canvas? Id duplication can break the functionality

File: src/components/Checkbox/Checkbox.js [hunk 3]
```diff
@@ -43,8 +50,8 @@ export default class Checkbox extends React.Component {
           }
         });
 
-    const required = <abbr class="chi-label__required" title="Required field">*</abbr>;
-    const optional = <abbr class="chi-label__optional" title="Optional field">(optional)</abbr>;
+    const required = <abbr className="chi-label__required" title="Required field">*</abbr>;
```
Comment (janibolkvadze) [inline src/components/Checkbox/Checkbox.js:53]: These classes are repeatedly used in different components. Can you add them to the classes.js, please?

File: src/components/Label/Label.js [hunk 4]
```diff
@@ -19,6 +22,19 @@ export default class Label extends React.Component {
       }
     }
 
+    const info = this.props.info
+      ? (
+        <div className="chi-label__help"
+        onClick={this.props.clickInfo}
+        onMouseEnter={this.props.mouseOverInfo}
+        onMouseLeave={this.props.mouseLeaveInfo}>
+          <chi-button id={"example__help-button"} type="icon" color="primary" size="sm" variant="flat" alternative-text="Help">
```
Comment (janibolkvadze) [inline src/components/Label/Label.js:31]: Potential ID duplication

File: src/components/Page-Title/Page-Title.js [hunk 5]
```diff
@@ -27,7 +27,7 @@ export default class PageTitle extends React.Component {
           </span>
         </div>
       </a>
-    ) : null;
+    ) : <a className="backlink-padding"></a>;
```
Comment (janibolkvadze) [inline src/components/Page-Title/Page-Title.js:30]: I understand this is necessary to reserve the blank space when the backlink is not provided. Can you add a custom class with fixed height, something like - `backlink-slot`or `backlink-blank`, instead, please?

File: src/components/Checkbox/Checkbox.js (hunk 2 above)
Comment (janibolkvadze) [reply src/components/Checkbox/Checkbox.js:31]: actually, why do we need the ID and data-target attributes if the designers are supposed to provide a custom interaction? And how do we trigger click, mouseDown, etc?

File: src/components/Label/Label.css [hunk 6]
```diff
@@ -0,0 +1,3 @@
+.chi .chi-label__help .chi-button.-flat.-icon:not(.-light) .chi-icon {
+    color: #0262b9!important;
```
Comment (janibolkvadze) [inline src/components/Label/Label.css:2]: Why is this necessary? Can not we use Chi color utility class here?
I believe, it's `-icon--primary`

File: src/components/Label/Label.js [hunk 7]
```diff
@@ -19,6 +22,19 @@ export default class Label extends React.Component {
       }
     }
 
+    const info = this.props.info
+      ? (
+        <div className="chi-label__help"
```
Comment (janibolkvadze) [inline src/components/Label/Label.js:27]: Let's add this className to classes.js as well, please

File: src/components/Number-input/Number-input.js [hunk 8]
```diff
@@ -95,6 +96,10 @@ NumberInput.propTypes = {
   mouseOver: PropTypes.func,
   mouseUp: PropTypes.func,
   valueChange: PropTypes.func,
+  info: PropTypes.bool,
+  clickInfo: PropTypes.func,
+  mouseDownInfo: PropTypes.func,
```
Comment (janibolkvadze) [inline src/components/Number-input/Number-input.js:101]: The approach of reusing <Label /> is cool. How do we apply these interactions to the icon button element inside Label? Does setting these interactions work?

File: src/components/Picker-group/Picker-group.js [hunk 9]
```diff
@@ -10,31 +10,43 @@ export default class PickerGroup extends React.Component {
...
 
     if (this.props.required && this.props.required === 'required') {
       message = required;
     } else if (this.props.required && this.props.required === 'optional')  {
       message = optional;
     }
+
+    const info = this.props.info ?
+    <div className="chi-label__help"  
+    onClick={this.props.clickInfo}
```
Comment (janibolkvadze) [inline src/components/Picker-group/Picker-group.js:25]: Indentation

File: src/components/Picker-group/Picker-group.js [hunk 10]
```diff
@@ -10,31 +10,43 @@ export default class PickerGroup extends React.Component {
...
     let message = '';
 
     if (this.props.required && this.props.required === 'required') {
       message = required;
     } else if (this.props.required && this.props.required === 'optional')  {
       message = optional;
     }
+
+    const info = this.props.info ?
+    <div className="chi-label__help"
```
Comment (janibolkvadze) [inline src/components/Picker-group/Picker-group.js:24]: classes.js

File: src/components/Picker-group/Picker-group.js [hunk 11]
```diff
@@ -10,31 +10,43 @@ export default class PickerGroup extends React.Component {
...
     } else if (this.props.required && this.props.required === 'optional')  {
       message = optional;
     }
+
+    const info = this.props.info ?
+    <div className="chi-label__help"  
+    onClick={this.props.clickInfo}
+    onMouseEnter={this.props.mouseOverInfo}
+    onMouseLeave={this.props.mouseLeaveInfo}>
+      <button className="chi-button -icon -sm -flat" id="example__help-button" aria-label="Help" data-target="#example__help-popover">
```
Comment (janibolkvadze) [inline src/components/Picker-group/Picker-group.js:28]:
```suggestion
      <button className="chi-button -icon -sm -flat" aria-label="Help">
```

File: src/components/Picker-group/Picker-group.js [hunk 12]
```diff
@@ -10,31 +10,43 @@ export default class PickerGroup extends React.Component {
   render() {
     const uuid = uuid4();
```
Comment (janibolkvadze) [inline src/components/Picker-group/Picker-group.js:11]: Is the 2+ Picker Groups issue fixed?

File: src/components/Radio/Radio.js [hunk 13]
```diff
@@ -53,19 +53,31 @@ export default class Radio extends React.Component {
...
     let message = '';
 
     if (this.props.required && this.props.required === 'required') {
       message = required;
     } else if (this.props.required && this.props.required === 'optional') {
       message = optional;
     }
+
+    const info = this.props.info ?
+    <div className="chi-label__help"
```
Comment (janibolkvadze) [inline src/components/Radio/Radio.js:67]: Same here, please

File: src/components/Text/Text.js [hunk 14]
```diff
@@ -23,9 +24,42 @@ export default class Text extends React.Component {
         break;
     }
 
+    switch (this.props.lineHeight) {
```
Comment (janibolkvadze) [inline src/components/Text/Text.js:27]: Is this necessary for adding the `default`option?

File: src/components/Text/Text.js [hunk 15]
```diff
@@ -23,9 +24,42 @@ export default class Text extends React.Component {
...
+        height = '8';
+        break;
+      case '72':
+        height = '9';
+        break;
+      default:
+        height = '3';
+        break;
+    }
+
```
Comment (janibolkvadze) [inline src/components/Text/Text.js:59]:
```suggestion
```

File: src/components/Text/Text.js [hunk 16]
```diff
@@ -23,9 +24,42 @@ export default class Text extends React.Component {
...
     return (
       <p className={`${this.props.size ? `-text--${this.props.size}` : ''}
-       ${this.props.lineHeight ? `-text -lh--${this.props.lineHeight / 8}` : ''}
+       ${this.props.lineHeight ? `-text -lh--${height}` : ''}
```
Comment (janibolkvadze) [inline src/components/Text/Text.js:62]:
```suggestion
       ${this.props.lineHeight ? `-lh--${lineHeight === 'default' ? 3 : lineHeight / 8}` : ''}
```

Comment (janibolkvadze) [reply src/components/Text/Text.js:62]: Something like this

**Review by janibolkvadze — CHANGES_REQUESTED** [review]
Some of the suggestions apply to other similar implementations. Please review adjust all the cases. Thanks!

File: src/components/Text/Text.js (hunk 14 above)
Comment (riya-abraham01) [reply src/components/Text/Text.js:27]: removed switch case

File: src/components/Checkbox/Checkbox.js (hunk 1 above)
Comment (riya-abraham01) [reply src/components/Checkbox/Checkbox.js:30]: fixed

File: src/components/Checkbox/Checkbox.js (hunk 2 above)
Comment (riya-abraham01) [reply src/components/Checkbox/Checkbox.js:31]: added interactions for click, mouseOver and mouseLeave

File: src/components/Picker-group/Picker-group.js (hunk 12 above)
Comment (riya-abraham01) [reply src/components/Picker-group/Picker-group.js:11]: not yet will work on it in another branch

File: src/components/Label/Label.js (hunk 7 above)
Comment (riya-abraham01) [reply src/components/Label/Label.js:27]: class name is already in STAT_CLASSES reused here

File: src/components/Page-Title/Page-Title.js (hunk 5 above)
Comment (riya-abraham01) [reply src/components/Page-Title/Page-Title.js:30]: <a className="chi-link -backlink-blank"></a>;

File: src/components/Number-input/Number-input.js (hunk 8 above)
Comment (riya-abraham01) [reply src/components/Number-input/Number-input.js:101]: added them separately

**Review by janibolkvadze — APPROVED** [review]
